client/src/components/LoginSession/LoginSessionContainer.js

```diff
@@ -0,0 +1,88 @@
+import PropTypes from 'prop-types';
+// import React, { Fragment } from 'react';
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import backend from 'lib/Backend';
+import Config from 'lib/Config'; // eslint-disable-line
+import LoginSession from './LoginSession';
+import { success, error } from 'state/toasts/ToastsActions';
+
+// Handle communication with  server on logout and toast notifications via redux
+// Using a class component rather than function component so that LoginSessionContainer-test.js
+// can use wrapper.instance() to test logout()
+class LoginSessionContainer extends Component {
```

Comment:
See the comment
```
Using a class component rather than function component so that LoginSessionContainer-test.js can use wrapper.instance() to test logout()
```
wrapper.instance() isn't available on functional components, it's the entire reason why i converted it to a class component.

https://enzymejs.github.io/enzyme/docs/api/ReactWrapper/instance.html
`With React 16 and above, instance() returns null for stateless functional components.`

Having hooks still doesn't seem to count for having state.  As soon as I converted it to a class component it worked

Reply:
I'll elaborate on the comment a bit more

Reply:
I suspect that's side effect of the bad split between your UI component and your the container.

Reply:
Nope, even after updating split as recommended, .instance() still doesn't work - https://github.com/silverstripe/silverstripe-session-manager/pull/47

Reply:
From my understanding we should be writing tests for hook components differently and not use the underlying instance, e.g. invoking methods on it etc.

See: https://reactjs.org/docs/hooks-faq.html#how-to-test-components-that-use-hooks
Basically, trigger click events etc like the user would, then check that the output matches what you'd expect.

I also ran into this problem when converting the `LoginSession` component.